src/amo/components/PermissionsCard/index.js

```diff
@@ -0,0 +1,280 @@
+import type { I18nType } from 'core/types/i18n';
+import Button from 'ui/components/Button';
+import Card from 'ui/components/Card';
+import Permission from 'ui/components/Permission';
+
+import './styles.scss';
+
+type Props = {|
+  addon: AddonType | null,
+  i18n: I18nType,
+  userAgentInfo: UserAgentInfoType,
+|};
+
+export class PermissionsCardBase extends React.Component<Props> {
+  render() {
+    const { addon, i18n, userAgentInfo } = this.props;
+
+    if (!addon) {
+      return null;
+    }
+
+    // These should be kept in sync with Firefox's strings for webextention permissions
+    // which can be found in
+    // https://hg.mozilla.org/mozilla-central/raw-file/tip/browser/locales/en-US/chrome/browser/browser.properties
+    const permissionStrings = {
+      bookmarks: i18n.gettext('Read and modify bookmarks'),
+      browserSettings: i18n.gettext('Read and modify browser settings'),
+      browsingData: i18n.gettext('Clear recent browsing history, cookies, and related data'),
+      clipboardRead: i18n.gettext('Get data from the clipboard'),
+      clipboardWrite: i18n.gettext('Input data to the clipboard'),
+      devtools: i18n.gettext('Extend developer tools to access your data in open tabs'),
+      downloads: i18n.gettext('Download files and read and modify the browser’s download history'),
+      'downloads.open': i18n.gettext('Open files downloaded to your computer'),
+      find: i18n.gettext('Read the text of all open tabs'),
+      geolocation: i18n.gettext('Access your location'),
+      history: i18n.gettext('Access browsing history'),
+      management: i18n.gettext('Monitor extension usage and manage themes'),
+      // In Firefox the following message replaces the name "Firefox" with the
+      // current brand name, e.g., "Nightly", but we do not need to do that.
+      nativeMessaging: i18n.gettext('Exchange messages with programs other than Firefox'),
+      notifications: i18n.gettext('Display notifications to you'),
+      pkcs11: i18n.gettext('Provide cryptographic authentication services'),
+      proxy: i18n.gettext('Control browser proxy setting'),
+      privacy: i18n.gettext('Read and modify privacy settings'),
+      sessions: i18n.gettext('Access recently closed tabs'),
+      tabs: i18n.gettext('Access browser tabs'),
+      tabHide: i18n.gettext('Hide and show browser tabs'),
+      topSites: i18n.gettext('Access browsing history'),
+      unlimitedStorage: i18n.gettext('Store unlimited amount of client-side data'),
+      webNavigation: i18n.gettext('Access browser activity during navigation'),
+      allUrls: i18n.gettext('Access your data for all websites'),
+      wildcard: (domain) => {
+        return i18n.sprintf(
+          i18n.gettext('Access your data for sites in the %(domain)s domain'),
+          { domain }
+        );
+      },
+      tooManyWildcards: (count) => {
+        return i18n.sprintf(
+          i18n.gettext('Access your data in %(count)s other domains'),
```

Comment:
The logic dictates that `count` will never be less than 2, which is why I didn't bother with pluralization. I suppose we could choose to change that logic down the road, in which case we'd be happy that we dealt with pluralization now, so that's an argument in favour of it, but I don't think we strictly need it right now.

Reply:
Ah, ok, then this should have a comment up above it so that some do-gooder developer doesn't change it one day thinking they caught a bug.